Realm/RLMObject.mm

```diff
@@ -40,7 +40,11 @@ - (instancetype)init {
 }
 
 - (instancetype)initWithValue:(id)value {
-    return [super initWithValue:value schema:RLMSchema.sharedSchema];
+    RLMSchema *schema = RLMSchema.partialSharedSchema;
+    if (![schema schemaForClassName:self.class.className]) {
```

Comment:
But then we'd still load the entire shared schema for a single class that's not in the partial shared schema. Seems like a mechanism to cache object schemas would be preferable then.